### PR TITLE
Wire coreApp into the local display manager for foreground/background arbitration

### PR DESCRIPTION
**Pull request by isaiahb**
## Problem

Running multiple miniapps on display glasses (G1/G2) is broken: any two apps that both want the display fight over it with no priority rule, because V2's `LocalDisplayManager` never learns which app is in the foreground.

`LocalDisplayManager` already implements the full V1-parity arbitration model as dead code:
- exactly one foreground ("core") app at a time
- exactly one background app can hold the display lock at a time (first-come-first-served)
- background apps can render *over* the foreground app
- when a background app clears or its request expires, the display reverts to the foreground app's own current request — only if it's still unexpired
- a lazy 10s-idle safety net lets another app steal a background app's slot on its next render attempt if the holder has gone quiet

None of this ever activates because `onCoreAppChange()` — the one entry point that tells the manager which app is foreground — was never called anywhere.

## Fix

**`coreApp` is derived state.** Instead of per-path call sites (which live testing proved to be whack-a-mole: the autostart boot-restore path bypasses `start()` entirely, and crash-respawns bypass everything), a store subscription reconciles the display-manager's core slot to "the currently-running standard app" on every store change. The store already mirrors JS-context liveness via the running-registry projection, so every path is covered: user tap, boot restore, crash-respawn, interop start, spawn failure, stopAll.

Live testing on-device surfaced two more model violations, fixed here:

**Boot text could strand.** Back-to-back mounts (exactly what the autostart restore loop does) snapshotted the previous boot's "Starting X…" as the pre-boot frame, and `endBoot` restored it as a permanent, expiry-less display ("Starting Merge…" stuck on the glasses). Chained boots now inherit the original pre-boot frame; `endBoot` restores/clears whenever the boot text is still on the glasses (including when every drained request lost arbitration); unmounting a mid-boot app does the same.

**Clear now forfeits the display.** A background app's clear (`render([])` or `clear_view`) was treated as another frame: it renewed the background lock with empty content, blanking the glasses while still blocking everyone else. A clear now means "I'm done": lock released, the core app's still-valid display restored, and the next requester can take the slot immediately.

## Verification

- `bun compile` clean; 24/24 `LocalDisplayManager` tests pass, including 6 new regression tests covering the three fixes; apps-store tests pass.
- On-device (Pixel 8 + Even Realities G2, dev build): app restart auto-restores Captions (standard) + Merge + Mentra AI (background); the reconciler claims Captions as core; chained boot text clears correctly; Mentra AI's "Hey Mentra" response renders over live captions and captions return afterward.

## Test plan

- [x] Foreground app (Captions) renders continuously; background app (Mentra AI) renders over it, then its frame expires/clears — captions' current content reappears.
- [x] App-restart restore path: restored foreground app claims the core slot without a user tap.
- [x] Back-to-back app mounts do not strand "Starting X…" on the glasses.
- [ ] Two background apps: second is blocked until the first clears or expires (unit-tested; not yet exercised on hardware).
- [ ] 10s-idle steal (unit-tested; not yet exercised on hardware).

---

> [!NOTE]
> **Medium Risk**
> Changes on-device glasses display arbitration and boot/clear edge cases; behavior is well covered by new unit tests but affects every local miniapp display path.
> 
> **Overview**
> Enables **foreground vs background display arbitration** on local miniapps by deriving the display manager’s core slot from the apps store: a subscription calls `onCoreAppChange` with the currently running standard app on every store update, so boot restore, crash-respawn, and user start/stop all claim or release core without per-path wiring.
> 
> **Boot “Starting…” handling** is tightened so chained mounts inherit the real pre-boot frame (not another boot message), `endBoot` restores or clears whenever boot text is still showing, and unmounting a booting app does the same—including when a clear queued during boot forfeited the prior frame.
> 
> **Clear / empty scene** is treated as **forfeit**, not a display frame: `isClearPayload` routes to `handleAppClear`, which drops background locks, clears core snapshots, and restores the core’s still-valid content instead of blanking the glasses while holding the lock. Core context teardown on unmount only drops `coreAppDisplay`, not the `coreApp` pointer, so crash-respawns keep core priority until the store says the app stopped.
> 
> Adds regression tests for boot stranding, clear-forfeit, and respawn core claim.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5e66a83bd6237c3c4221be33e4db99168dc08c56. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: mobile/modules/engine/src/stores/apps.ts [hunk 1]
```diff
@@ -406,6 +413,13 @@ export const useAppStatusStore = create<AppStatusState>((set, get) => ({
       return
     }
 
+    // Release the display-manager's foreground slot. The foreground-only-one
```
[the remark this answers is not in the document]
Comment (isaiahb) [reply mobile/modules/engine/src/stores/apps.ts]: Fixed in ad47f07: the spawn-failure path now releases the claim, skipping the release if another standard app started and claimed the slot while this spawn was in flight.

File: mobile/modules/engine/src/stores/apps.ts [hunk 2]
```diff
@@ -406,6 +413,13 @@ export const useAppStatusStore = create<AppStatusState>((set, get) => ({
...
+    // invariant means at most one standard app is ever running, so this is
+    // always releasing the app that currently holds it.
+    if (app.type === "standard") {
+      localDisplayManager.onCoreAppChange(null)
+    }
```
[the remark this answers is not in the document]
Comment (isaiahb) [reply mobile/modules/engine/src/stores/apps.ts]: Fixed in ad47f07: stop() now only releases the slot for a running standard app, which under the foreground-only-one invariant is always the current holder.

File: mobile/modules/engine/src/services/LocalDisplayManager.ts [hunk 3]
```diff
@@ -349,6 +381,33 @@ class LocalDisplayManager {
...
+    if (this.currentDisplay?.packageName === packageName) {
+      this.clearExpiryTimer()
+      this.currentDisplay = null
+      this.tryRestoreCoreDisplay()
+      if (!this.currentDisplay) {
+        this.sendClear()
+      }
+    }
+    resolve?.({status: "displayed"})
+  }
```
[the remark this answers is not in the document]
Comment (isaiahb) [reply mobile/modules/engine/src/services/LocalDisplayManager.ts:430]: Confirmed and fixed in ded6d8c: both restore sites (endBoot and the unmount-mid-boot path) now skip a pre-boot frame whose owner's final boot-queued intent was a clear, so a forfeit during the boot window cannot be resurrected. Regression tests added for both sites.

File: mobile/modules/engine/src/services/LocalDisplayManager.ts [hunk 4]
```diff
@@ -211,9 +223,19 @@ class LocalDisplayManager {
...
     if (this.bootingApp?.packageName === packageName) {
+      const preBoot = this.bootingApp.preBoot
       this.cancelBoot()
+      if (this.currentDisplay?.packageName === SYSTEM_BOOT_PKG) {
+        if (preBoot && (preBoot.expiresAt === null || preBoot.expiresAt > this.now())) {
+          this.restoreDisplay(preBoot)
+        } else {
+          this.sendClear()
+        }
+      }
```
[the remark this answers is not in the document]
Comment (isaiahb) [reply mobile/modules/engine/src/services/LocalDisplayManager.ts:242]: Confirmed and fixed in ded6d8c: onUnmount no longer writes coreApp at all — the pointer is derived state owned by the apps-store reconciler, and the liveness/crash respawn path (runtime.unregisterApp, no registry/store change) was exactly the flow that left it stale. onUnmount now only drops the dead context's saved frame; the claim survives the respawn. Regression test covers the teardown-then-respawn flow. Note the finding referenced the true-pong watchdog from 8099cea, which was moved off this PR — but the gap applied equally to the existing missed-pings watchdog, so the fix stands on its own.
